src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -667,3 +664,122 @@ export const testFailedInit = {
 // - [x] test-zlib-failed-init.js
 // - [ ] test-zlib-invalid-input.js
 // - [ ] test-zlib-reset-before-write.js
+
+const BIG_DATA = 'horse'.repeat(50_000) + 'cow'.repeat(49_000);
+
+export const inflateSyncTest = {
+  test() {
+    strictEqual(
+      zlib.inflateSync(zlib.deflateSync(BIG_DATA)).toString(),
+      BIG_DATA
+    );
+
+    throws(
+      () => zlib.inflateSync('garbage data'),
+      Error('incorrect header check')
+    );
+
+    strictEqual(
+      zlib
+        .inflateSync(Buffer.from('OE9LyixKUUiCEQAmfgUk', 'base64'), {
```

Comment:
let's include test variations that test a zero-length input and an overly large input (zip bomb style)?